### DISCUSSION TITLE
Should we switch the Core tests from Google Test to Qt Test?

### DISCUSSION THREAD

**Discussion by nbolton** [deskflow/deskflow#7807] [Polls]
Chris (@sithlord48) has suggested that we may want to switch from [Google Test](https://github.com/google/googletest) to [Qt Test](https://doc.qt.io/qt-6/qtest-overview.html). From his perspective, he believes tests written in Qt Test are easier to understand compared to Google Test (this view could be biased; if your favorite tool is a hammer, everything looks like a nail). Also, we have been discussing potentially depending more on Qt in the Core (we currently only use Qt D-Bus in the Core).

Since this project was started around 2 decades ago (as Synergy), we've used Google Test and as a team we are most familiar with that. With Google Test, we currently have 10,000 out of 60,000 lines of code covered which is gradually increasing. Switching to Qt Test would mean abandoning our current coverage and starting from scratch, essentially re-implementing 10,000 lines of code coverage which is a huge undertaking.

However, just because we're using Google Test for the Core, doesn't mean we can't _also_ use Qt Test for the GUI. If we were to continue writing tests with Google Code (rather than abandoning) then we could write new tests just for the GUI using Qt Test.

One thing to note is that we are also considering to use [Squish](https://www.qt.io/product/quality-assurance/squish) to test the Qt GUI as well as Google Code and/or Qt Test.

**Comment by sithlord48** [comment]
I would port all the codes 
I want to do this to increase where we are using Qt. There are many places we can benefit from using Q Classes and moving to Qt Testing would happen as I move items to Qt. 

some issues would just not be a problem any more since Qt provides easy ways to do them https://github.com/deskflow/deskflow/issues/7803, https://github.com/deskflow/deskflow/issues/7784, https://github.com/deskflow/deskflow/issues/7732 to name a few I also can remove alot of the native platform code and use Qt's abstractions ones tested better then any we will write...
